src/loaders/ObjectLoader.js

```diff
@@ -26,11 +28,17 @@ import { MaterialLoader } from './MaterialLoader';
 import { BufferGeometryLoader } from './BufferGeometryLoader';
 import { JSONLoader } from './JSONLoader';
 import { XHRLoader } from './XHRLoader';
+import * as CoreGeometry from '../core/Geometries';
+import * as ExtraGeometry from '../extras/geometries/Geometries';
 
 /**
  * @author mrdoob / http://mrdoob.com/
  */
 
+function resolveGeometryType ( type ) {
```

Comment:
Haha, I wouldn't call it so much over-engineered, as I'd say general stupidity on my own behalf! :wink:

I have a bad tendency when submitting pull requests to try make things operate similarly to their prior implementation, even when it no longer makes any sense!

I'll sort it out shortly.